### PR TITLE
Spot bugs configuration

### PR DESCRIPTION
**Pull request by manticore-projects**
Resolve all SpotBugs P1 issues in Main and Test
Add Spot Bugs Filter File, which focuses on P1 issues only 
Configure Spot Bug Task to stop on any new P1 issue

Next steps would be to work on P2 issues.

### CODE REVIEW

File: poi-examples/src/main/java/org/apache/poi/examples/hssf/usermodel/AddDimensionedImage.java [hunk 1]
```diff
@@ -18,6 +18,10 @@ Licensed to the Apache Software Foundation (ASF) under one or more
 
 package org.apache.poi.examples.hssf.usermodel;
 
-/* Placeholder - this is now handled in the Common SS example **/
+/**
+ * Placeholder Class - this is now handled in the Common SS example
+ * @deprecated in 5.1.0 - use {@link org.apache.poi.examples.ss.AddDimensionedImage}
+ */
```
Comment (pjfanning) [inline poi-examples/src/main/java/org/apache/poi/examples/hssf/usermodel/AddDimensionedImage.java:26]: can you add `@Removal(version="6.0.0")`?

File: poi-ooxml/src/main/java/org/apache/poi/poifs/crypt/temp/EncryptedTempData.java [hunk 2]
```diff
@@ -46,8 +46,9 @@
  */
 @Beta
 public class EncryptedTempData {
-    private static Logger LOG = LogManager.getLogger(EncryptedTempData.class);
+    public static final SecureRandom RANDOM = new SecureRandom();
```
Comment (pjfanning) [inline poi-ooxml/src/main/java/org/apache/poi/poifs/crypt/temp/EncryptedTempData.java]: can you move this somewhere else? shouldn't be public in this `@Beta` internal class

File: poi-ooxml/src/main/java/org/apache/poi/poifs/crypt/temp/EncryptedTempData.java [hunk 3]
```diff
@@ -46,8 +46,9 @@
...
 
+    private static Logger LOG = LogManager.getLogger(EncryptedTempData.class);
```
Comment (pjfanning) [inline poi-ooxml/src/main/java/org/apache/poi/poifs/crypt/temp/EncryptedTempData.java]: this should be final

File: poi-scratchpad/src/test/java/org/apache/poi/hslf/usermodel/TestPictures.java [hunk 4]
```diff
@@ -64,6 +64,7 @@ Licensed to the Apache Software Foundation (ASF) under one or more
  */
 public final class TestPictures {
     private static final POIDataSamples slTests = POIDataSamples.getSlideShowInstance();
+    public final static SecureRandom RANDOM = new SecureRandom();
```
Comment (pjfanning) [inline poi-scratchpad/src/test/java/org/apache/poi/hslf/usermodel/TestPictures.java]: why public?

File: poi/src/test/java/org/apache/poi/ss/formula/function/ExcelFileFormatDocFunctionExtractor.java [hunk 5]
```diff
@@ -31,14 +31,7 @@ Licensed to the Apache Software Foundation (ASF) under one or more
...
 import java.security.MessageDigest;
-import java.util.ArrayList;
-import java.util.Arrays;
-import java.util.HashMap;
-import java.util.HashSet;
-import java.util.List;
-import java.util.Map;
-import java.util.Set;
-import java.util.Stack;
+import java.util.*;
```
Comment (pjfanning) [inline poi/src/test/java/org/apache/poi/ss/formula/function/ExcelFileFormatDocFunctionExtractor.java]: can you avoid * imports?

**Comment by manticore-projects** [comment]
Thank you so much for reviewing and feedback.
I have fixed those and especially have moved all calls to Random into a Singleton.

**Comment by pjfanning** [comment]
Thanks - merged - just one issue: the class with the SecureRandom was missing a license comment, that is fixed now

**Comment by manticore-projects** [comment]
Thanks for helping out, I am still a learner in this regard. Next time will be better.
Cheers

**Pull request closed without merging**